src/liballoc/raw_vec.rs

```diff
@@ -235,7 +235,60 @@ impl<T, A: Alloc> RawVec<T, A> {
         }
     }
 
-    /// Doubles the size of the type's backing allocation. This is common enough
+    /// Grows the vector capacity by `capacity_increase`.
+    ///
+    /// It allows implementing amortized O(1) `push` on vector-like containers.
+    ///
+    /// # Attributes
+    ///
+    /// - `#[inline]`: LLVM is able to completely elide memory allocations in
+    ///   many cases if it can "see" where memory is allocated and freed.
+    /// - `#[cold]`: calling this function is a "rare" event
+    ///
+    #[inline]
+    #[cold]
+    pub fn grow_by(&mut self, capacity_increase: usize) {
+        let elem_size = mem::size_of::<T>();
+        assert!(elem_size != 0, "RawVecs of zero-sized types can't grow");
+
+        let (new_cap, uniq) = match self.current_layout() {
+            Some(cur) => {
+                // The invariant `elem_size * self.cap <= isize::MAX` is
+                // maintained by `alloc_guard`; the alignment will never be too
+                // large as to "not be specifiable" (so we can use
+                // `from_size_align_unchecked`).
+                let new_cap = Self::suitable_capacity(self.cap, capacity_increase);
+                let new_size = new_cap * elem_size;
+                let new_layout = unsafe { Layout::from_size_align_unchecked(new_size, cur.align()) };
```

Comment:
These lines are too long.

```
[00:04:18] tidy error: /checkout/src/liballoc/raw_vec.rs:262: line longer than 100 chars
[00:04:18] tidy error: /checkout/src/liballoc/raw_vec.rs:264: line longer than 100 chars
[00:04:18] tidy error: /checkout/src/liballoc/raw_vec.rs:354: trailing whitespace
[00:04:18] tidy error: /checkout/src/liballoc/raw_vec.rs:378: line longer than 100 chars
[00:04:18] tidy error: /checkout/src/liballoc/raw_vec.rs:380: line longer than 100 chars
[00:04:18] tidy error: /checkout/src/liballoc/raw_vec.rs:457: line longer than 100 chars
```

Reply:
@kennytm i've fixed these except the two links, AFAIK rustdoc does not support spliting links over multiple lines

Reply:
@gnzlbg You could use the syntax:

```markdown
blah blah blah [link][x] blah blah blah

[x]: https://www.example.com/extremely/long/link?can=use&syntax=like#this
```

Reply:
@kenny but what happens when the link is longer than 100 chars ? How do I break it so that the docs still work ?

Reply:
@gnzlbg `tidy` will ignore long lines in that form. Not sure about rustfmt.